src/components/global/Task/Task.jsx

```diff
@@ -1,16 +1,81 @@
 import React, { useState } from "react";
+import { Draggable } from "react-beautiful-dnd";
 import RightChevron from "../../../assets/icons/RightChevron";
+import TaskBoardSampleData from "../TaskBoardSampleData";
 import Subtask from "./Subtask";
 
 function Task(props) {
-  const { name, checked, expanded, onClick, selected, subtasks } = props;
+  const {
+    id,
+    name,
+    index,
+    checked,
+    expanded,
+    subtasks,
+    selected,
+    onClick,
+    boardTask,
+  } = props;
 
   const [isChecked, setIsChecked] = useState(checked);
   const [isExpanded, setIsExpanded] = useState(expanded);
 
-  const handleCheckBoxClick = () => setIsChecked(!isChecked);
+  const handleCheckBoxClick = () => {
+    if (boardTask) {
+      // if we're on the trello board, update the file that contains the task data
+      TaskBoardSampleData.tasks[boardTask.id].checked = !isChecked;
+    }
+    setIsChecked(!isChecked);
+  };
   const handleIconClick = () => setIsExpanded(!isExpanded);
 
+  if (boardTask) {
```

Comment:
We are rendering something completely different based on one flag, so this should be a separate component. If you want to reuse the styling you can, just use the same css class names as the original task component and structure the divs the same way.

Reply:
Components have been separated